src/main/java/net/snowflake/client/core/SFBaseSession.java

```diff
@@ -202,93 +219,201 @@ public void setSessionId(String sessionId) {
     this.sessionId = sessionId;
   }
 
+  /**
+   * @return true if session is in SQLMode
+   */
   public boolean isSfSQLMode() {
     return sfSQLMode;
   }
 
+  /**
+   * Set sfSQLMode
+   *
+   * @param sfSQLMode boolean
+   */
   public void setSfSQLMode(boolean sfSQLMode) {
     this.sfSQLMode = sfSQLMode;
   }
 
+  /**
+   * Get the database version
+   *
+   * @return database version
+   */
   public String getDatabaseVersion() {
     return databaseVersion;
   }
 
+  /**
+   * Set database version
+   *
+   * @param databaseVersion the version to set
+   */
   public void setDatabaseVersion(String databaseVersion) {
     this.databaseVersion = databaseVersion;
   }
 
+  /**
+   * Get databse major version
+   *
+   * @return the database major version
+   */
   public int getDatabaseMajorVersion() {
     return databaseMajorVersion;
   }
 
+  /**
+   * Set database major version
+   *
+   * @param databaseMajorVersion the database major version
+   */
   public void setDatabaseMajorVersion(int databaseMajorVersion) {
     this.databaseMajorVersion = databaseMajorVersion;
   }
 
+  /**
+   * Get the database minor version
+   *
+   * @return database minor version
+   */
   public int getDatabaseMinorVersion() {
     return databaseMinorVersion;
   }
 
+  /**
+   * Set the database minor version
+   *
+   * @param databaseMinorVersion the minor version
+   */
   public void setDatabaseMinorVersion(int databaseMinorVersion) {
     this.databaseMinorVersion = databaseMinorVersion;
   }
 
+  /**
+   * Get if prepared statement logging is enabled
+   *
+   * @return true if enabled
+   */
   public boolean getPreparedStatementLogging() {
     return this.preparedStatementLogging;
   }
 
+  /**
+   * Set prepared statement logging
+   *
+   * @param value boolean
+   */
   public void setPreparedStatementLogging(boolean value) {
     this.preparedStatementLogging = value;
   }
 
+  /**
+   * Get inject file upload failure
+   *
+   * @return file to fail
```

Comment:
for the inject sth methods let's add information that they should be used only in internal tests